tests/e2e/demo-modal-compare.spec.ts

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable no-console -- we want to log when test fails */
+import { isTesting as isTestingEnvironment } from '@lmc-eu/spirit-common/constants/environments';
+import { SERVERS, getDevelopmentEndpointUri } from '@lmc-eu/spirit-common/constants/servers';
+import { expect, test } from '@playwright/test';
+
+const runComponentCompareTests = ({ componentsDir, packageName, componentName }) => {
+  if (!packageName) return;
+
+  const formattedPackageName = packageName
+    .split('-')
+    .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+    .join(' ');
+
+  test.describe(`Demo ${formattedPackageName} Components`, () => {
+    test(`test demo ${formattedPackageName} component ${componentName}`, async ({ page }) => {
+      try {
+        const url = isTestingEnvironment()
+          ? SERVERS.TESTING[packageName]
+          : getDevelopmentEndpointUri(packageName, { isDocker: packageName !== 'web-twig' });
```

Comment:
A lot of the same code as elsewhere, so maybe we should abstract it somehow?

Reply:
made some helper functions